src/SFA.DAS.Reservations.Application.UnitTests/Reservations/Commands/CreateReservationLevyEmployer/WhenCreatingANewReservationForLevyEmployer.cs

```diff
@@ -0,0 +1,87 @@
+    {
+        [Test, MoqAutoData]
+        public void ThenTheCommandGetsValidatedAndThrowsAnExceptionIfNotValid(
+            CreateReservationLevyEmployerCommand request,
+            [Frozen] Mock<IValidator<CreateReservationLevyEmployerCommand>> validator,
+            CreateReservationLevyEmployerCommandHandler handler)
+        {
+            //Arrange
+            validator.Setup(x => x.ValidateAsync(It.IsAny<CreateReservationLevyEmployerCommand>()))
+                .ReturnsAsync(new ValidationResult {ValidationDictionary = new Dictionary<string, string> {{"", ""}}});
+            //Act
+            Assert.ThrowsAsync<ValidationException>(async() => await handler.Handle(request, CancellationToken.None));
+
+            //Assert
+            validator.Verify(x => x.ValidateAsync(request), Times.Once());
+            
+        }
+
+        [Test, MoqAutoData]
+        public async Task ThenCallsReservationServiceToCreateReservation(
+            [Frozen]CreateReservationLevyEmployerCommand request,
+            [Frozen] Mock<IValidator<CreateReservationLevyEmployerCommand>> validator,
+            [Frozen] Mock<IReservationService> service,
+            CreateReservationLevyEmployerCommandHandler handler)
+        {
+            //Arrange
+            validator.Setup(x => x.ValidateAsync(request))
+                .ReturnsAsync(new ValidationResult {ValidationDictionary = new Dictionary<string, string>()});
+
+            service.Setup(x =>
+                    x.CreateReservationLevyEmployer(It.IsAny<Guid>(), request.AccountId, request.AccountLegalEntityId))
+                .ReturnsAsync(new CreateReservationResponse());
+
+            //Act
+            var result = await handler.Handle(request, CancellationToken.None);
+            
+            //Assert
+            service.Verify( x => x.CreateReservationLevyEmployer(It.IsAny<Guid>(), request.AccountId, request.AccountLegalEntityId));
+
+        }
+
+        [Test, MoqAutoData]
+        public async Task ThenReturnsCorrectResult(
+            [Frozen] Guid id,
+            [Frozen]CreateReservationLevyEmployerCommand request,
+            [Frozen] Mock<IValidator<CreateReservationLevyEmployerCommand>> validator,
+            [Frozen] Mock<IReservationService> service,
+            CreateReservationLevyEmployerCommandHandler handler)
+        {
+            //Arrange
+            validator.Setup(x => x.ValidateAsync(request))
+                .ReturnsAsync(new ValidationResult { ValidationDictionary = new Dictionary<string, string>() });
+
+            service.Setup(x =>
+                    x.CreateReservationLevyEmployer(It.IsAny<Guid>(), request.AccountId, request.AccountLegalEntityId))
+                .ReturnsAsync(new CreateReservationResponse(){Id = id});
+
+            //Act
+            var result = await handler.Handle(request, CancellationToken.None);
+
+            //Assert
+            Assert.AreEqual(id,result.ReservationId);
```

Comment:
The assertion of this test can be moved up to the other test as the behaviour is covered there